### PR TITLE
splitting domain model to the command side(decider) and query side (view)

### PR DESCRIPTION
**Pull request by idugalic**
The Domain layer `Timer.kt` is split into Decide and View. This way we decouple the decision-making process from how the view state should be presented on the screen.

The `TimerViewState` belongs to the Domain layer and FModel View is used to evolve it out of the same events published by the Decider. Notice how this TimerViewState is modeled to match your requirements of how the screen should look. That is the point of CQRS, to denormalize the state, so it matches many different requirements. 

In `MainActivity` we additionally map TimerViewState to `@Stable TimerViewStateUI`

_The Demo application might be too small to demonstrate the benefits of CQRS._

### CODE REVIEW

File: app/src/main/java/fraktal/io/ext/Reducer.kt [hunk 1]
```diff
@@ -26,38 +23,40 @@ fun <C, S, E> Reducer(
...
+    private val _commands: Channel<C> = Channel(capacity = capacity)
+    private val _events: Channel<E?> = Channel(capacity = capacity)
+
+    val uiStates = _viewState.asStateFlow()
     val events = _events.consumeAsFlow()
 
     init {
         scope.launch {
-            commandChannel.consumeAsFlow().collectLatest { command ->
+            // Only Collect LATEST is supported ???
```
Comment (jQwout) [inline app/src/main/java/fraktal/io/ext/Reducer.kt:40]: To be honest, this is a very controversial decision. related to whether we want to maintain a queue of commands or not. I can't answer unequivocally

File: app/src/main/java/fraktal/io/ext/Reducer.kt [hunk 2]
```diff
@@ -26,38 +23,40 @@ fun <C, S, E> Reducer(
...
     }
 
     suspend fun emit(command: C) {
-        commandChannel.send(command)
+        _commands.send(command)
     }
 
     private suspend fun handle(command: C) {
-        val decidedEvents = decider.decide.invoke(command, _states.value)
+        val decidedEvents = decider.decide.invoke(command, _deciderState.value)
```
Comment (jQwout) [inline app/src/main/java/fraktal/io/ext/Reducer.kt:52]: cool. but it looks like we don't need the state of Desider at all. Okay. I, in principle, like this option, but.

 I'm probably far from a functional approach, I meant that we can subscribe to changes to the domain state and observe it, then somehow merge it into the ui state. But, you have done state tracking, and based on it we calculate the ui state.
 
 as a result, we calculate the state twice based on the event is TimerEvent.OnTimerTick(in TimerDecider and . What can be inconvenient in other cases if this calculation is large and complex?
 
 can you please explain why it is important to do this, or how you can avoid it?

File: app/src/main/java/fraktal/io/ext/Reducer.kt (hunk 1 above)
Comment (idugalic) [reply app/src/main/java/fraktal/io/ext/Reducer.kt:40]: OK. I wanted to highlight this, as in some situations you might want to do this differently and not cancel the flow with `collectLatest`

File: app/src/main/java/fraktal/io/ext/Reducer.kt (hunk 2 above)
Comment (idugalic) [reply app/src/main/java/fraktal/io/ext/Reducer.kt:52]: CQRS is all about creating a separate domain model for the View state(s) = Query Model(s). That is the reason you have that second state (`TimerViewState`). This is the way! :) The first state is just for the Decider to enable its decision-making process (publishing events).  This demo is too small/simple in scope to see the benefits of that. 

`TimerViewState` state and any other View state that you might create are built out of Events, and these Events play the central role in this design. Observe how `TimerViewState` is modeled/structured, to fully support the rendering of that View (one one-on-one). 

You usually want to think about that Query model early in your Design. Event Modeling is a good fit to create that blueprint. See example here [https://github.com/fraktalio/fmodel-ktor-demo/blob/main/.assets/restaurant-model2.jpg](https://github.com/fraktalio/fmodel-ktor-demo/blob/main/.assets/restaurant-model2.jpg). Observe how Green sticky notes represent different View States. 

You don't avoid this. You would like to start using it. It is a great benefit that will allow you NOT to depend on a single State (of your decider) to rule them all.  That schema/state will never fit the need!

Comment (idugalic) [reply app/src/main/java/fraktal/io/ext/Reducer.kt:52]: This is not related to the Functional paradigm, but rather tries to remove the limitation of using single schema/state to serve the needs of the Presentation layer. On the Server Side we sometimes go that far that each Web Page has its own table/view/state (just for that page in the Web Flow of many pages)

Comment (idugalic) [reply app/src/main/java/fraktal/io/ext/Reducer.kt:52]: True, the Reducer is a View Reducer and does not need the Decider state `TimerState`. I ll look to remove it.

**Pull request merged**